### PR TITLE
change want/have to only negotiate one module/capability at once

### PR DESCRIPTION
**Pull request by majewsky**
This vastly simplifies implementations. The previous format looks nice on paper, but requires an absurd amount of bookkeeping on the server side.

### CODE REVIEW

File: spec/core/1.0.md [hunk 1]
```diff
@@ -352,100 +352,111 @@ A server **agrees to** using a module or capability by answering a `want` messag
...
+Upon receiving a valid `want` message, the server MUST reply with a `have` message to indicate if it agrees to using the requested module or capability.
+If the `want` message is valid, but the server does not agree to the requested module and capability, it MUST reply with a `have` message with zero arguments.
 
-For `want` arguments that are pairs of module names and major version, the corresponding `have` argument is the original `want` argument, with a dot and a positive integer appended to it.
-The positive integer indicates the maximal minor version of this module that is understood by the server.
-Receipt of the `have` message containing this argument entitles the client to use all message types, properties and capabilities defined in a specification of that module with the same major version number and the same, or a smaller, minor version number.
+If the `want` message requests usage of a module and the server agrees to using that module, it MUST reply with a `have` message with three arguments: the module name, the major and minor version of the specification supported by the server.
```
Comment (laerling) [inline spec/core/1.0.md]: Why not `full-version`?

File: spec/core/1.0.md [hunk 2]
```diff
@@ -352,100 +352,111 @@ A server **agrees to** using a module or capability by answering a `want` messag
...
-
-Each argument in the `have` message corresponds to an argument in the `want` message, except that modules and capabilities to which the server does not agree are omitted.
-This particularly means that the argument list of the `have` message MUST preserve the order of arguments in the `want` message.
-From this and the definition of the `want` message also follows that the use of a module must be agreed to before any of its capabilities can be agreed to.
-
-For `want` arguments that are capability names, the corresponding `have` argument is the same string, the capability name.
-A server SHALL NOT agree to using a capability if it has not previously agreed to using the module defining said capability, unless it agrees to using the module in the same `have` message.
+Upon receiving a valid `want` message, the server MUST reply with a `have` message to indicate if it agrees to using the requested module or capability.
+If the `want` message is valid, but the server does not agree to the requested module and capability, it MUST reply with a `have` message with zero arguments.
```
Comment (laerling) [inline spec/core/1.0.md]: s/and/or/

File: spec/core/1.0.md [hunk 3]
```diff
@@ -352,100 +352,111 @@ A server **agrees to** using a module or capability by answering a `want` messag
...
-
-```vt6
-(want core1 foo2)
-(have core1.0 foo2.3)
---> valid; client can now use core1.0 module and either foo2.0, foo2.1, foo2.2 or foo2.3 module
-
-(want core1 core2)
-(have core2.1)
---> valid; client can now use either core2.0 or core2.1 module, but not core1.x modules
+If the `want` message requests usage of a capability and the server agrees to using that module, it MUST reply with a `have` message with a single argument: the capability name as accepted by `<scoped-name>`.
```
Comment (laerling) [inline spec/core/1.0.md]: s/module/capability/

File: spec/core/1.0.md [hunk 4]
```diff
@@ -352,100 +352,111 @@ A server **agrees to** using a module or capability by answering a `want` messag
...
+Receipt of this `have` message entitles the client to use all message types, properties and behavior defined in a specification of that module with the same major version number and the same, or a smaller, minor version number, where such functionality depends on the negotiated capability.
```
Comment (laerling) [inline spec/core/1.0.md]: Why should the client be entitled to use everything of a module or anything from the module in the first place? Also, the second half sentence concerns capabilities, while the first one concerns modules. Split that into two sentences.

File: spec/core/1.0.md [hunk 5]
```diff
@@ -352,100 +352,111 @@ A server **agrees to** using a module or capability by answering a `want` messag
...
 
-(want core1 foo.cap)
-(have)
---> "want" is invalid; cannot negotiate capability from foo1 before foo1 module has been agreed upon
+A server SHALL NOT agree to using a capability if it did not agree to using a module that defines this capability in any prior `have` message on the same server connection.
 
-(want core1 foo1 foo.cap)
-(have core1.0 foo1.1)
---> valid; server has not agreed to capability
+When a `want` message concerns the same capability as an earlier `want` message, the server MUST reply consistently.
```
Comment (laerling) [inline spec/core/1.0.md]: Duplicate of line 405, although this time with explanation (line 415).

File: spec/core/1.0.md [hunk 6]
```diff
@@ -352,100 +352,111 @@ A server **agrees to** using a module or capability by answering a `want` messag
...
+The major and minor version MUST be formatted according to the `<major-version>` and `<minor-version>` grammar elements defined above.
+The major version in the `have` message MUST be one of the major versions that appeared in the original `want` message.
 
+The server MUST NOT agree to using a module if it does not support any of the major versions of that module which were offered in the `want` message.
 The server MUST NOT agree to using multiple major versions of the same module, neither in a single `have` message or across multiple `have` messages on the same server connection.
 The server MAY agree to using different major versions of the same module on different server connections.
 
-When an argument in a `want` message is identical to an earlier argument in the same `want` message, or to an argument in an earlier `want` message, the server MUST reply consistently.
-If it did not agree to the earlier argument, it MUST NOT agree to this argument, and vice versa.
+When a `want` message concerns the same module as an earlier `want` message, the server MUST reply consistently.
```
Comment (laerling) [inline spec/core/1.0.md:407]: Duplicate of line 414.

**Review by laerling — CHANGES_REQUESTED** [review]

File: spec/core/1.0.md (hunk 3 above)
Comment (majewsky) [reply spec/core/1.0.md]: No, 405 is about modules. 414 is about capabilities.

File: spec/core/1.0.md (hunk 4 above)
Comment (majewsky) [reply spec/core/1.0.md]: No, you're misunderstanding this convoluted sentence. This paragraph is only about negotiating capabilities.

File: spec/core/1.0.md (hunk 5 above)
Comment (majewsky) [reply spec/core/1.0.md]: No. See above.

File: spec/core/1.0.md (hunk 6 above)
Comment (majewsky) [reply spec/core/1.0.md:407]: No. See above.

File: spec/core/1.0.md (hunk 3 above)
Comment (laerling) [reply spec/core/1.0.md]: Then better write "the module that defines that capability", or something like that.

**Comment by hkgit03** [comment]
>Riot, pls.

**Review by laerling — APPROVED** [review]

**Pull request merged**